pkg/controller/controller_utils.go

```diff
@@ -1048,6 +1048,9 @@ func AddOrUpdateTaintOnNode(ctx context.Context, c clientset.Interface, nodeName
 		if err != nil {
 			return err
 		}
+		if oldNode == nil {
```

Comment:
err in line 1048 would be non-nil and return when node doesn't exist

Reply:
Got it. The problem is not here. 

https://github.com/kubernetes/kubernetes/blob/ccf57ba09dafa94906310059e42a6b896c0eaf44/pkg/controller/testutil/test_utils.go#L136-L156

The fake node handler should behave like the real server. It should return some errors when getting a node that does not exist.

Updated.